### PR TITLE
Include instruction data in contexts

### PR DESCRIPTION
**Pull request by Dodecahedr0x**
Solves #24 and enables pushing #22 forward.

I chose to create a new `args` field in every context to make these context arguments accessible.

### CODE REVIEW

**Review by Aursen — COMMENTED** [review]
Seems good to me, but what will happen if the structure is directly the args structure?

**Comment by Aursen** [comment]
> Solves #24 and enables pushing #22 forward.
> 
> I chose to create a new `args` field in every context to make these context arguments accessible.

It was my my goal too because you cannot work with another tuple here

**Comment by Dodecahedr0x** [comment]
> Seems good to me, but what will happen if the structure is directly the args structure?

Do you mean what happens in the transfer-sol example? In this case it's available directly.

What isn't tested here though is to have multiple contexts with arguments, or using the instruction attribute along with args as handler parameters. Will add some tests cases for this

File: crates/context-macro/src/lib.rs [hunk 1]
```diff
@@ -69,16 +83,34 @@ impl Parse for Context {
...
+            Item::Struct(account_struct) => {
+                // Add an `args` field to the context
+                if let Fields::Named(fields) = &mut account_struct.fields {
+                    fields.named.push(parse_quote! {
+                        pub args: crayfish_context::args::Args<#new_lifetime, #args_struct_name>
+                    });
+                }
+
+                // Remove attributes
+                account_struct.attrs.clear();
```
Comment (Aursen) [inline crates/context-macro/src/lib.rs:105]: Here only remove the args attribute to allow flexibility

**Review by Aursen — CHANGES_REQUESTED** [review]

**Comment by Dodecahedr0x** [comment]
Thanks for the feedback, the last commit addresses all these issues

**Review by Aursen — APPROVED** [review]
LFG

**Pull request merged**